### PR TITLE
Better parse error handling

### PR DESCRIPTION
**Pull request by hhhjort**

### CODE REVIEW

**Review by guscarreon — DISMISSED** [review]
LGTM

File: constants/constants.go [hunk 1]
```diff
@@ -13,3 +13,6 @@ const (
 	SectionUSPUT  SectionID = 11
 	SectionUSPCT  SectionID = 12
 )
+
+var SectionIDNames = []string{"ID0", "ID1", "tcfeu2", "gpp header", "ID4", "ID5", "uspv1", "uspnat",
+	"uspca", "uspva", "uspco", "usput", "uspct"}
```
Comment (SyntaxNode) [inline constants/constants.go]: IMHO: It's a bit odd to see names set for the value 0 which is undefined. There are names for sections 1, 4, and 5 - but they're not supported by this library. Based on how the function is used, it's save to ignore values for which are not parsed.

How do you feel about using a map for supported sections?

```
var sectionNamesByID = map[int]string{
  2: "tcfeu2",
  3: "gpp header",
  ...
}
```

File: parse.go [hunk 2]
```diff
@@ -73,37 +73,64 @@ func Parse(v string) (GppContainer, error) {
...
 
-	return gpp, nil
+	return gpp, errs
+}
+
+func sectionParseError(id int, err error) error {
+	if id < len(constants.SectionIDNames) && id > 0 {
+		return fmt.Errorf("Error parsing %s consent string: %s", constants.SectionIDNames[id], err)
+	} else {
+		return fmt.Errorf("Error parsing UNKNOWN consent string: %s", err)
```
Comment (SyntaxNode) [inline parse.go]: Perhaps it's better to say unsupported for future compatibility and include the section id? Something like:
```
fmt.Errorf("error parsing unsupported (section %d) consent string: %s", err)
```

Comment (SyntaxNode) [inline parse.go]: Nitpick: Please start with lower case "error" for errors in this func.

File: parse_test.go [hunk 3]
```diff
@@ -113,12 +115,34 @@ func TestParse(t *testing.T) {
...
+			gppString:   "DBGBMA~CPXxRfAPXxRfAAfKABENB-CgAAAAAAAAAAYgAAAAAAAA",
+			expected: GppContainer{
+				Version:      1,
+				SectionTypes: []constants.SectionID{2},
+				Sections: []Section{GenericSection{sectionID: 2,
+					value: "CPXxRfAPXxRfAAfKABENB-CgAAAAAAAAAAYgAAAAAAAA"}},
+			},
+			expectedError: []error{fmt.Errorf("error parsing GPP header, section identifiers: error reading an int offset value in a Range(Fibonacci) entry(1): error reading bit 4 of Integer(Fibonacci): expected 1 bit at bit 32, but the byte array was only 4 bytes long")},
+		},
+		"GPP-uspca-error": {
```
Comment (SyntaxNode) [inline parse_test.go:129]: Nice naming. 👍

File: parse.go (hunk 2 above)
Comment (hhhjort) [reply parse.go]: No problem

Comment (hhhjort) [reply parse.go]: Sounds good. I think I was overthinking things with this helper function anyway. :)

File: constants/constants.go (hunk 1 above)
Comment (hhhjort) [reply constants/constants.go]: Works for me.

File: parse.go (hunk 2 above)
Comment (SyntaxNode) [reply parse.go]: :)

**Review by SyntaxNode — APPROVED** [review]

**Review by guscarreon — APPROVED** [review]
LGTM

**Pull request merged**